sympy/solvers/ode.py

```diff
@@ -1351,8 +1352,40 @@ class in it.  Note that a hint may do this anyway if
                             not check.has(zoo) and not check.has(-oo):
                             coeff_dict = {'p': p, 'q': q, 'x0': point, 'terms': terms}
                             matching_hints["2nd_power_series_regular"] = coeff_dict
-
-
+                            # If the ODE has regular singular point at x0 and is of the form
+                            # Eq((x)**2*Derivative(y(x),x,x)+x*Derivative(y(x),x)+(a4**2*x**2-n**2)*y(x) thus Bessel's equation
+                            if (p.is_constant(x) and p != 0) and r[c3] != 0:
+                                a4 = Wild('a4', exclude=[x,f(x),df])
+                                b4 = Wild('b4', exclude=[x,f(x),df])
+                                c4 = Wild('c4', exclude=[x,f(x),df])
+                                d4 = Wild('d4', exclude=[x,f(x),df])
+                                coeff = r[a3].match(a4*x**2)
+                                if coeff:
+                                    coeff = coeff[a4]
+                                    if not coeff.has(x):
+                                        r[c3] = r[c3]/coeff
+                                        r[b3] = r[b3]/coeff
+                                        r[a3] = r[a3]/coeff
+                                    rn = r[c3].match(a4*a4*x**(2*d4)-b4*b4)
+                                    if check==0: # if r[c3] becomes zero at x0
+                                        rn = r[c3].match(a4*a4*x**(2*d4))
+                                        if rn:
+                                            rn[b4] = 0
+                                    if rn and r[b3] != 0:
+                                        rn = {'n':rn[b4], 'a4':rn[a4], 'd4':rn[d4]}
+                                        rn['c4'] = r[b3].match(c4*x)[c4]
+                                        matching_hints["2nd_linear_bessel"] = rn
```

Comment:
Can we put this matching code into a separate function like `_2nd_linear_bessel_match`?

Reply:
We can but for that we need to check that ode is ordinary or not and after that we have to check the singularity point and all of these are checking in above code for others matching(for 2nd order ode). So if we will make a new function, we have to do all these things again.